modules/data_release/ajax/AddPermission.php

```diff
@@ -12,25 +12,92 @@
  *  @link     https://github.com/aces/Loris
  */
 
-$DB =& Database::singleton();
+$user     = \User::singleton();
+$factory  = \NDB_Factory::singleton();
+$settings = $factory->settings();
+$baseURL  = $settings->getBaseURL();
+$DB       = $factory->database();
```

Comment:
This should probably have a permissions check.

Reply:
Permission checks are below

Reply:
As indicated by @johnsaigle, lines 18 to 21 could be included in the `if` block and executed only when required. Not a show stopper.

Reply:
they would need to be in the if and the else if... duplication